client/foundries.go

```diff
@@ -542,10 +542,58 @@ func readResponse(res *http.Response, log logrus.FieldLogger) (*[]byte, error) {
 		// Still return a body, a caller might need it, but also return an error
 		msg := fmt.Sprintf("HTTP error during %s '%s': %s",
 			res.Request.Method, res.Request.URL.String(), res.Status)
-		if len(body) < PRINT_LIMIT {
-			// return an error response body up to a meaningful limit - if it spans beyond a few
-			// lines, need to find a more appropriate message.
-			msg = fmt.Sprintf("%s\n=%s", msg, body)
+
+		// Some APIs return well-formatted errors, try to use them
+		var (
+			useGenericError bool
+			listErrors      struct {
+				Msg     string   `json:"msg,omitempty"`
+				Message string   `json:"message,omitempty"`
+				Errors  []string `json:"errors,omitempty"`
+			}
+			dictErrors struct {
+				Msg     string            `json:"msg,omitempty"`
+				Message string            `json:"message,omitempty"`
+				Errors  map[string]string `json:"errors,omitempty"`
+			}
+		)
+
+		if merr := json.Unmarshal(body, &listErrors); merr == nil {
+			if listErrors.Msg != "" {
```

Comment:
Just really not important comment. The first part for these two types of Errors is the same so maybe the code can be unified.

Reply:
My first thought was about doing it in a functional way :).